sharding/collator.go

```diff
@@ -40,8 +40,19 @@ func subscribeBlockHeaders(c collatorClient) error {
 		// Query the current state to see if we are an eligible proposer
 		log.Info(fmt.Sprintf("Received new header: %v", head.Number.String()))
 		// TODO: Only run this code on certain periods?
-		if err := checkShardsForProposal(c, head); err != nil {
-			return fmt.Errorf("unable to watch shards. %v", err)
+
+		// Check if we are in the validator pool before checking if we are an eligible proposer
+		v, err := checkForValidators(c)
+		if err != nil {
+			return fmt.Errorf("unable to verify client in validator pool", err)
+		}
+
+		if (v) {
+			if err := checkShardsForProposal(c, head); err != nil {
+				return fmt.Errorf("unable to watch shards. %v", err)
+			}
+		} else {
+			log.Warn("client not in validator pool, re-try next block")
```

Comment:
Please replace client with account. Capitalize sentence. Optional: Add account hash to log message.

`log.Warn(fmt.Sprintf("Account %s not in validator pool.", account.Hash))`

Reply:
do you mean account.Address?

Reply:
yes